Add ArgoCD issues notes in the official documentation (#20313)
cilium-team/cilium: Documentation/configuration/argocd-issues.rst

.. only:: not (epub or latex or html)

    WARNING: You are looking at unreleased Cilium documentation.
    Please use the official rendered version released here:
    https://docs.cilium.io

.. _argocd_issues:

***********************************
Issues Deploying Cilium with ArgoCD
***********************************

> There have been a lot of reports from users hitting issues with ArgoCD. This documentation 

Reviewer: “A lot of” is not precise and doesn't bring any info here.

```suggestion
There have been reports from users hitting issues with ArgoCD. This documentation 
```

page outlines some of the known issues.

Known issues with solutions
===========================
The following are some known issues with ArgoCD that have solutions. Click on them to 
see their solutions on GitHub.

> - `ArgoCD deletes CustomResourceDefinitions that Cilium uses to implement networking and security <https://github.com/cilium/cilium/issues/17349>`__.
> - `ciliumid not found  <https://github.com/cilium/cilium/issues/17614>`__.
> - `Out-of-sync issues for cilium using ArgoCD <https://github.com/cilium/cilium/issues/14550>`__.
> - `CiliumEndpoint missing for a pod <https://github.com/cilium/cilium/issues/17047>`__.
> - `hubble-relay: failed to create gRPC client <https://github.com/cilium/cilium/issues/16361>`__.
> - `Add a configmap checksum to automatically roll daemonset <https://github.com/cilium/cilium/issues/14331>`__.
> - `Cilium does not recognise identity thus got policy denied <https://github.com/cilium/cilium/issues/14284>`__.
> - `Intermittent 504 error from service when using ingress network policy <https://github.com/cilium/cilium/issues/13240>`__.

Reviewer: I'd use this (but it's your call)

```suggestion
- ArgoCD deletes CustomResourceDefinitions that Cilium uses to implement networking and security (:gh-issue:`17349`)
- ciliumid not found  (:gh-issue:`17614`)
- Out-of-sync issues for cilium using ArgoCD (:gh-issue:`14550`)
- CiliumEndpoint missing for a pod (:gh-issue:`17047`)
- hubble-relay: failed to create gRPC client (:gh-issue:`16361`)
- Add a configmap checksum to automatically roll daemonset (:gh-issue:`14331`)
- Cilium does not recognise identity thus got policy denied (:gh-issue:`14284`)
- Intermittent 504 error from service when using ingress network policy (:gh-issue:`13240`)
```

Also maybe sort the issues either by number or alphabetically?

.. note::
    
    If you have an issue with ArgoCD that's not outlined above, check this 

>     `list of ArgoCD-related issues on GitHub <https://github.com/cilium/cilium/issues?q=is%3Aissue+argocd+is%3Aclosed>`__. 

Reviewer: Why the `is:closed`? I understand you don't want open issues _only_, but if you remove this filter, you'd get both open and closed issues. Wouldn't that be better? Open issues may not have a workaround yet, but sometimes they do, or sometimes it's also helpful to see that the issues has been reported somewhere already?

Author: Yes true. Sorry, that's a mistake on my end. Meant to put it without the filter. Will make the edit.

    If you can't find an issue that relates to yours, create one and/or seek help on the :term:`Slack channel`. 

>     With Cilium contributors across the globe, someone is almost always available to help.

Reviewer: ```suggestion
```

I'd remove this - We could have that in the community section, but we don't want to expand on the channel every time we link to it in the docs.

Author: Thanks for this feedback.

.. only:: not (epub or latex or html)

    WARNING: You are looking at unreleased Cilium documentation.
    Please use the official rendered version released here:
    https://docs.cilium.io

.. _argocd_issues:

************************************
Issues Deploying Cilium with Argo CD

> ************************************

Reviewer: I'd like these to be renamed, so that it no longer contains `issues`. My fear is that reading `Issues ... with Argo CD` from the index of the docs will read like it's not well supported and will scare users away from trying it at all.

Not sure what title would be best, maybe `Troubleshooting Cilium deployed with Argo CD` or `Workarounds/Adjustments/Additional Steps for Deploying...` or just `Deploying ...`, or at least `Known Issues and their Workarounds for Deploying...`. Or if you have anything better :).

Author: Very true. Thanks for pointing this out. I think `Troubleshooting Cilium deployed with Argo CD` is perfect. 

Thank you.

There have been reports from users hitting issues with Argo CD. This documentation 
page outlines some of the known issues and their solutions.

> Argo CD deletes CustomResourceDefinitions that Cilium uses to implement networking and security

Reviewer: Nit: This is a long title, maybe `Argo CD deletes CustomResourceDefinitions` and make it more explicit if necessary in the description that comes below?

Author: That makes sense. Will do so. Thank you.

===============================================================================================

When deploying Cilium with Argo CD, some users have reported that Cilium-generated custom resources disappear,
causing one or more of the following issues:

- ``ciliumid`` not found (:gh-issue:`17614`)
- Argo CD Out-of-sync issues for hubble-generate-certs (:gh-issue:`14550`)
- Out-of-sync issues for Cilium using Argo CD (:gh-issue:`18298`)

Solution
--------

To prevent these issues, declare resource exclusions in the Argo CD ``ConfigMap`` by following `these instructions <https://argoproj.github.io/argo-cd/operator-manual/declarative-setup/#resource-exclusioninclusion>`__.

Here is an example snippet:

.. code-block:: yaml

    resource.exclusions: |
     - apiGroups:
         - cilium.io
       kinds:
         - CiliumIdentity
       clusters:
         - "*"

> Also, on Slack, a user reported that the problem affects all workloads you deploy with Argo CD in a cluster running Cilium, not just Cilium itself.

Reviewer: Nit: No need for `on Slack`.

```suggestion
Also, it has been reported that the problem may affect all workloads you deploy with Argo CD in a cluster running Cilium, not just Cilium itself.
```

Author: Done. Thank you.
